### PR TITLE
Ensure all paths and config are setup before app launch.

### LINKED ISSUE #1027: Kernels not connecting when missing `nteract.json` file
While trying to get the install up on `master`, my kernel wasn't actually connecting. Popping up the dev tools showed that `~/Library/Jupyter/nteract.json` was missing. After creating the file with an empty object in it, the kernel connected just fine.

I would suggest that the file be automatically created if it doesn't exist, but I don't know the implications to the rest of the system.

cc @rgbkrk and @jdetle who helped diagnose the issue on slack.

### PR DESCRIPTION
**Pull request by rgbkrk**
Fixes #1027, #804.

### CODE REVIEW

File: src/main/index.js [hunk 1]
```diff
@@ -53,7 +61,37 @@ const fullAppReady$ = Rx.Observable.zip(
...
+      mkdirpObservable(jupyterConfigDir)
+    )
+  )
+  // Set up our configuration file
+  .mergeMap(() =>
+    readFileObservable(nteractConfigFilename)
+      .catch((err) => {
+        if (err.code === 'ENOENT') {
+          return writeFileObservable(nteractConfigFilename, JSON.stringify({
+            // TODO: Pull the default config from somewhere
```
Comment (captainsafia) [inline src/main/index.js:83]: Might also wanna include the sendMetrics in this config.

**Review by captainsafia — CHANGES_REQUESTED** [review]
Looks great!

File: src/main/index.js (hunk 1 above)
Comment (rgbkrk) [reply src/main/index.js:83]: Setting it to false for now?

Comment (rgbkrk) [reply src/main/index.js:83]: I mostly left it out to provide flexibility for you later. Otherwise how will you prompt for it in the future?

Comment (captainsafia) [reply src/main/index.js:83]: Yep! False by default. In the immediate future we can do the dialog box that asks the user what they want like before. And in the far future, we would have it as part of the config UI I outlined in another issue somewhere....

Comment (rgbkrk) [reply src/main/index.js:83]: If I leave it `undefined` or set it to `null`, when you implement the metrics setup again you can check for nulliness so we're not missing all the early adopters.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**